services/meta/service_test.go

```diff
@@ -1397,3 +1447,17 @@ func mustMarshalJSON(v interface{}) string {
 	}
 	return string(b)
 }
+
+func freePort() string {
```

Comment:
Could `net.Listen` ever return an error (and therefore `l` would be `nil`)? I guess this maybe doesn't matter in the tests, but you have some `defers` after `freePort` calls which wouldn't get run if there was a `nil` pointer `panic`. 

Just a thought!